src/main/java/net/patchworkmc/patcher/util/ResourceDownloadUtil.java

```diff
@@ -0,0 +1,169 @@
+package net.patchworkmc.patcher.util;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.UncheckedIOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.List;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import org.apache.commons.io.FileUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import net.fabricmc.stitch.merge.JarMerger;
+import net.fabricmc.tinyremapper.IMappingProvider;
+import net.fabricmc.tinyremapper.TinyUtils;
+
+import net.patchworkmc.patcher.Patchwork;
+import net.patchworkmc.patcher.mapping.BridgedMappings;
+import net.patchworkmc.patcher.mapping.RawMapping;
+import net.patchworkmc.patcher.mapping.TinyWriter;
+import net.patchworkmc.patcher.mapping.Tsrg;
+import net.patchworkmc.patcher.mapping.TsrgClass;
+import net.patchworkmc.patcher.mapping.TsrgMappings;
+
+public class ResourceDownloadUtil {
+	private static final String FORGE_MAVEN = "https://files.minecraftforge.net/maven";
+	private static final String FABRIC_MAVEN = "https://maven.fabricmc.net";
+	private final Path tempDir;
+
+	private static final Logger LOGGER = LogManager.getLogger(ResourceDownloadUtil.class);
+	private IMappingProvider srg;
+	private IMappingProvider bridged;
+
+	public ResourceDownloadUtil() {
+		try {
+			tempDir = Files.createTempDirectory(new File(System.getProperty("java.io.tmpdir")).toPath(), "pw-p-ResourceDownloadUtil-");
```

Comment:
Why is a separate new temporary directory needed here? It's probably possible to reuse the main temporary directory

Reply:
Recycling someone else's temp directory means we have to add a lot of safety checks since we can't promise the tempDir is empty, unlike with this approach